ibm/resource_ibm_dl_gateway_virtual_connection.go

```diff
@@ -0,0 +1,279 @@
+				Required:     true,
+				ForceNew:     false,
+				ValidateFunc: validateRegexpLen(1, 63, "^([a-zA-Z]|[a-zA-Z][-_a-zA-Z0-9]*[a-zA-Z0-9])$"),
+				Description:  "The type of virtual connection.Allowable values",
+			},
+			dlVCNetworkId: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "Unique identifier of the target network. For type=vpc virtual connections this is the CRN of the target VPC. This field does not apply to type=classic connections.",
+			},
+			dlVCCreatedAt: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			dlVCStatus: {
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: false,
+			},
+
+			dlVCNetworkAccount: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+func resourceIBMdlGatewayVCValidator() *ResourceValidator {
+
+	validateSchema := make([]ValidateSchema, 1)
+	vcType := "classic, vpc"
+	validateSchema = append(validateSchema,
+		ValidateSchema{
+			Identifier:                 dlVCType,
+			ValidateFunctionIdentifier: ValidateAllowedStringValue,
+			Type:                       TypeString,
+			Required:                   true,
+			AllowedValues:              vcType})
+	ibmDLGatewayVCResourceValidator := ResourceValidator{ResourceName: "ibm_dl_gateway_vc", Schema: validateSchema}
+	return &ibmDLGatewayVCResourceValidator
+}
+func resourceIBMdlGatewayVCCreate(d *schema.ResourceData, meta interface{}) error {
+	directLink, err := directlinkClient(meta)
+	if err != nil {
+		return err
+	}
+
+	createGatewayVCOptions := &directlinkapisv1.CreateGatewayVirtualConnectionOptions{}
+
+	gatewayId := d.Get(dlGatewayId).(string)
+	createGatewayVCOptions.SetGatewayID(gatewayId)
+	vcName := d.Get(dlVCName).(string)
+	createGatewayVCOptions.SetName(vcName)
+	vcType := d.Get(dlVCType).(string)
+	createGatewayVCOptions.SetType(vcType)
+
+	if _, ok := d.GetOk(dlVCNetworkId); ok {
+		vcNetworkId := d.Get(dlVCNetworkId).(string)
+		createGatewayVCOptions.SetNetworkID(vcNetworkId)
+	}
+
+	gatewayVC, response, err := directLink.CreateGatewayVirtualConnection(createGatewayVCOptions)
+	if err != nil {
+		log.Printf("[DEBUG] Create Direct Link Gateway (Dedicated) Virtual connection err %s\n%s", err, response)
+		return err
+	}
+	d.SetId(*gatewayVC.ID)
+
+	log.Printf("[INFO] Created Direct Link Gateway (Dedicated Template) Virtual connection : %s", *gatewayVC.ID)
+
+	if err != nil {
+		log.Printf("Error creating Direct Link Gateway (Dedicated Template) Virtual Connection :%s", response)
+		return err
+	}
+
+	return resourceIBMdlGatewayVCRead(d, meta)
+}
+
+func resourceIBMdlGatewayVCRead(d *schema.ResourceData, meta interface{}) error {
+
+	directLink, err := directlinkClient(meta)
+	if err != nil {
+		return err
+	}
+
+	ID := d.Id()
+
+	getGatewayVirtualConnectionOptions := &directlinkapisv1.GetGatewayVirtualConnectionOptions{}
+	gatewayId := d.Get(dlGatewayId).(string)
```

Comment:
Since ID is combination of two IDS use 
```
parts, err := idParts(d.Id())
	if err != nil {
		return err
	}

	gatewayId := parts[0]
	ID := parts[1]
```

Don't use d.Get(XXX) in Read method

Use the simialr Split in all methods (Read, update, delete, exists)